MaskRCNN_no_batch_convergence/train.py

```diff
@@ -363,26 +364,23 @@ def rpn(self, image, features, inputs):
         else:
             losses = []
 
-        return BoxProposals(proposal_boxes), losses
+        return proposal_boxes, losses
 
-    def roi_heads(self, image, features, proposals, targets):
+    def roi_heads(self, image, features, proposal_boxes, targets):
         image_shape2d = tf.shape(image)[2:]     # h,w
         assert len(features) == 5, "Features have to be P23456!"
         gt_boxes, gt_labels, *_ = targets
 
-        if self.training:
-
-
-
+        #############################################################################################################
+        if BATCH_FRCNN_HEAD:
+        #############################################################################################################
+            prepadding_gt_counts = tf.expand_dims(tf.shape(gt_labels)[0], axis=0)  # 1 x NumGT
```

Comment:
I've moved this here because it was being used in two blocks: BATCH_SAMPLE_TARGETS and BATCH_FAST_RCNN_LOSSES